relayer/relays/beacon/header/header.go

```diff
@@ -42,71 +41,46 @@ func (h *Header) Sync(ctx context.Context, eg *errgroup.Group) error {
 	if err != nil {
 		return fmt.Errorf("fetch parachain last finalized header state: %w", err)
 	}
-	latestSyncedPeriod := h.syncer.ComputeSyncPeriodAtSlot(lastFinalizedHeaderState.BeaconSlot)
 
+	latestSyncedPeriod := h.syncer.ComputeSyncPeriodAtSlot(lastFinalizedHeaderState.BeaconSlot)
+	log.WithFields(log.Fields{
+		"hash":   lastFinalizedHeaderState.BeaconBlockRoot,
+		"slot":   lastFinalizedHeaderState.BeaconSlot,
+		"period": latestSyncedPeriod,
+	}).Info("set cache: last finalized state")
 	h.cache.SetLastSyncedSyncCommitteePeriod(latestSyncedPeriod)
-
-	log.WithField("period", latestSyncedPeriod).Info("set cache: last beacon synced sync committee period")
-
+	h.cache.SetLastSyncedFinalizedState(lastFinalizedHeaderState.BeaconBlockRoot, lastFinalizedHeaderState.BeaconSlot)
 	h.cache.AddCheckPointSlots([]uint64{lastFinalizedHeaderState.BeaconSlot})
 
-	log.WithField("finalizedSlots", h.cache.Finalized.Checkpoints.Slots).Info("set cache: finalized checkpoint slots")
-
-	currentFinalizedHeader, err := h.syncer.GetLatestFinalizedHeader()
-	if err != nil {
-		return fmt.Errorf("fetch last finalized header state from beacon node: %w", err)
-	}
-
-	err = h.syncLaggingSyncCommitteePeriods(ctx, latestSyncedPeriod, uint64(currentFinalizedHeader.Payload.FinalizedHeader.Slot))
+	// syncLaggingExecutionHeaders so to allow ExecutionHeader to catch up
+	executionHeaderState, err := h.writer.GetLastExecutionHeaderState()
 	if err != nil {
-		return fmt.Errorf("sync lagging sync committee periods: %w", err)
+		return fmt.Errorf("fetch last execution hash: %w", err)
 	}
```

Comment:
no need `syncLaggingSyncCommitteePeriods` as we'll do it later in `SyncHeadersFromFinalized`